sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -216,16 +220,23 @@ private Flux<List<StoreResult>> readFromReplicas(List<StoreResult> resultCollect
             return Flux.error(new GoneException());
         }
         List<Pair<Flux<StoreResponse>, Uri>> readStoreTasks = new ArrayList<>();
+        Map<Uri, String> replicaStatusesAttempting = new HashMap<>();
+        Map<Uri, String> replicaStatusesNotAttempting = new HashMap<>();
 
+        resolveApiResults.forEach(uri -> {
+            replicaStatusesNotAttempting.put(uri, uri.getHealthStatusDiagnosticString());
+        });
         List<Uri> addressRandomPermutation = AddressEnumerator.getTransportAddresses(entity, resolveApiResults);
 
         // The health status of the Uri will change as the time goes by
         // what we really want to track is the health status snapshot at this moment
-        List<String> replicaStatusList =
-                addressRandomPermutation
-                        .stream()
-                        .map(uri -> uri.getHealthStatusDiagnosticString())
-                        .collect(Collectors.toList());
+        addressRandomPermutation.forEach(uri -> {
+            replicaStatusesNotAttempting.remove(uri);
+            replicaStatusesAttempting.put(uri, uri.getHealthStatusDiagnosticString());
+        });
+        Map<String, List<String>> replicaStatusList = new HashMap<>();
+        replicaStatusList.put(Uri.ATTEMPTING ,new ArrayList<>(replicaStatusesAttempting.values()));
```

Comment:
Is it necessary to allocate a new list and copy the values? Why not just add replicaStatusesAttemtping.values()?

Reply:
I was doing it convert into a list because .values returns a collection. Decided to change it to a set instead of using a map and a list.